tests/playwright-test/golden.spec.ts

```diff
@@ -96,6 +176,52 @@ test('should update expectations', async ({runInlineTest}, testInfo) => {
   expect(data.toString()).toBe('Hello world updated');
 });
 
+test('should update expectations with negated matcher', async ({runInlineTest}, testInfo) => {
+  const result = await runInlineTest({
+    'a.spec.js-snapshots/snapshot.txt': `Hello world`,
+    'a.spec.js': `
+      const { test } = pwt;
+      test('is a test', ({}) => {
+        expect('Hello world updated').not.toMatchSnapshot('snapshot.txt');
+      });
+    `
+  }, { 'update-snapshots': true });
+  expect(result.exitCode).toBe(0);
+  expect(result.output).toContain('snapshot.txt does not match, writing actual.');
+  const data = fs.readFileSync(testInfo.outputPath('a.spec.js-snapshots/snapshot.txt'));
+  expect(data.toString()).toBe('Hello world updated');
+});
+
+test('should silently write missing expectations locally with the update-snapshots flag', async ({runInlineTest}, testInfo) => {
+  const result = await runInlineTest({
+    'a.spec.js': `
+      const { test } = pwt;
+      test('is a test', ({}) => {
+        expect('Hello world').toMatchSnapshot('snapshot.txt');
+      });
+    `
+  }, { 'update-snapshots': true }, { CI: '' });
+  expect(result.exitCode).toBe(0);
+  expect(result.output).toContain('snapshot.txt is missing in snapshots, writing actual');
+  const data = fs.readFileSync(testInfo.outputPath('a.spec.js-snapshots/snapshot.txt'));
+  expect(data.toString()).toBe('Hello world');
+});
+
+test('should silently write missing expectations locally with the update-snapshots flag with negate matcher', async ({runInlineTest}, testInfo) => {
+  const result = await runInlineTest({
+    'a.spec.js': `
+      const { test } = pwt;
+      test('is a test', ({}) => {
+        expect('Hello world').not.toMatchSnapshot('snapshot.txt');
```

Comment:
Same here, I am not sure this is the right behavior.

Reply:
It's a new one, I've unswerd in the [thread](https://github.com/microsoft/playwright/pull/7345/files/fe01d1df92b0eabea96b8ce3baf38432f29231b5#r661722006). But I'm going to change the logic, it'll fail if snapshot is missing.